storage/admin_storage.go

```diff
@@ -91,4 +91,11 @@ type AdminWriter interface {
 	// Remaining fields must be set to valid values.
 	// Returns an error if the tree is invalid or creation fails.
 	CreateTree(ctx context.Context, tree *trillian.Tree) (*trillian.Tree, error)
+
+	// UpdateTree updates the specified tree in storage, returning a tree
+	// with all storage-generated fields set.
+	// updateFunc is called to perform the desired tree modifications.
```

Comment:
I consider `trillian.Tree` the canonical source of what's readonly/mutable/valid/etc. I've added a comment here to that effect. I prefer not to list the fields here too, so I don't have to worry about updating documentation in more than one place when trillian.Tree changes.